working/0546-patterns/patterns-feature-specification.md

```diff
@@ -2054,11 +2087,18 @@ fail in some way.*
         expression's type is `dynamic`.* If it evaluates to `false`, continue to
         the next case (or default or exit).
 
-    3.  Execute the nearest non-empty case body at or following this case.
-        *You're allowed to have multiple empty cases where all preceding
-        ones share the same body with the last case.*
+    3.  Find the nearest non-empty case body at or following this case. *You're
+        allowed to have multiple empty cases where all preceding ones share the
+        same body with the last case.*
+
+    4.  Initialize any consistent variables declared in the body scope with the
+        values of the corresponding variables from the case scope. *There will
+        be no consistent variables and the body scope will be the case scope if
```

Comment:
"be no inconsistent variables"?

Can we *not* make it the case scope, because if we do, we can't make the variables final in the case scope (which I want to).

Reply:
No, I meant "no consistent". If there aren't shared cases, the there are no consistent variables that are declared as distinct from the variables in the case pattern itself and that need to get initialized. Instead, the body refers directly to the actual variables from the pattern.

> Can we not make it the case scope, because if we do, we can't make the variables final in the case scope (which I want to).

Done. I did this deliberately because I intended to match parameter/function body scope. But it turns out I misremembered how function body scopes worked. :)

I want these two to behave the same:

```dart
test(int x) {
  var x = 2;
}

switch (0) {
  case int x: var x = 2;
}
```

Either both of these should be an error, or neither should be. I thought the former was an error so specified that the body uses the same scope as the scope where the pattern variables are declared, but I was wrong. There's no error in the former so there should be no error in the latter. Tweaked the proposal to state that the body has its own scope whose enclosing scope is the case body scope.